### PR TITLE
[SPARK-35358][BUILD] Increase maximum Java heap used for release build to avoid OOM

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch proposes to increase the maximum heap memory setting for release build.

### Why are the changes needed?

When I was cutting RCs for 2.4.8, I frequently encountered OOM during building using mvn. It happens many times until I increased the heap memory setting.

I am not sure if other release managers encounter the same issue. So I propose to increase the heap memory setting and see if it looks good for others.

### Does this PR introduce _any_ user-facing change?

No, dev only.

### How was this patch tested?

Manually used it during cutting RCs of 2.4.8.

### CODE REVIEW

**Comment by viirya** [comment]
cc @HyukjinKwon @dongjoon-hyun

File: dev/create-release/release-build.sh [hunk 1]
```diff
@@ -210,6 +210,8 @@ if [[ "$1" == "package" ]]; then
     PYSPARK_VERSION=`echo "$SPARK_VERSION" |  sed -e "s/-/./" -e "s/SNAPSHOT/dev0/" -e "s/preview/dev/"`
     echo "__version__='$PYSPARK_VERSION'" > python/pyspark/version.py
 
+    export MAVEN_OPTS="-Xmx12000m"
```
Comment (dongjoon-hyun) [inline dev/create-release/release-build.sh:213]: Can we have this globally outside of `if` statement? Then, it looks like we need only one line addition.

Comment (viirya) [reply dev/create-release/release-build.sh:213]: okay

Comment (maropu) [reply dev/create-release/release-build.sh:213]: nit: we can say `-Xmx12g`?

Comment (viirya) [reply dev/create-release/release-build.sh:213]: ok.

**Comment by srowen** [comment]
Getting pretty big! but OK if needed.

**Comment by viirya** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master/3.1/3.0.

**Pull request closed without merging**